File: REST-Server/src/main/java/pl/antonic/server/methods/ImageProcessingMethodWrapper.java
package pl.antonic.server.methods;

import java.util.List;

public class ImageProcessingMethodWrapper {

    private List<ImageProcessingMethod> methods;

    public List<ImageProcessingMethod> getMethods() {
        return methods;
    }

    public boolean contains(Methods method) {
        for (ImageProcessingMethod imageProcessingMethod : methods) {
            if (imageProcessingMethod.getId() == method)
                return true;
        }

        return false;
    }

    public void setMethods(List<ImageProcessingMethod> methods) {
        this.methods = methods;
    }
}

File: AndroidCV/app/src/main/java/pl/antonic/androidcv/methods/ImageProcessingMethodWrapper.kt
package pl.antonic.androidcv.methods

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class ImageProcessingMethodWrapper : Serializable {

    @SerializedName("methods")
    @Expose
    val methods = mutableListOf<ImageProcessingMethod>()

    fun containsId(id: Methods) : Boolean {
        for (method in methods) {
            if (method.id == id)
                return true
        }

        return false
    }

    fun getMethod(id: Methods) : ImageProcessingMethod? {
        for (method in methods) {
            if (method.id == id)
                return method
        }

        return null
    }
}